.gitattributes

```diff
@@ -0,0 +1,8 @@
+# Set the default behavior, in case people don't have core.autocrlf set.
+* text=auto
```

Comment:
I never use Windows, but why do we want any conversion of line endings at all? It seems like it would be be better to just have them always be LF at all times.

Reply:
I guess you have `core.autocrlf` enabled?

I think we can disable any line ending conversion with:

```
* -text
```

Reply:
It seems risky without that safety net. It is a major pain in the ass to revert CRLF to LF manually if any get introduced (git diffs don't really show the distinction of different line endings - so I've been bogged down by this in the past).

I read yesterday (on the git docs) that the value `input` would convert CRLF to LF ("but not the other way around"). The docs are terribly worded to the point where the config option's explanation is actually confusing.

Reply:
It is not too hard to convert line endings if you find a problem --- e.g. you can use the dos2unix program.

I think the way you currently have it, autocrlf behavior will apply to all files by default if the user has `core.autocrlf` enabled, except that `.py`, `.rst`, and `.scss` files will always have `LF` line endings when checked out (even on Windows).

I'm not sure why we want some files to have autocrlf mode and some files always LF.

Reply:
well, I did some experimenting.
- Setting my global config `core.autocrlf=input` provides LF endings in the local workspace on checkout. But this overrides the `eol` config option.
- When `eol` is set to `lf`, then the workspace uses LF endings (which is the desired result - even on Windows), but this means `autocrlf` should be set to `false` (which means no normalizaton of line endings on checkout).
- If `core.autocrlf=false` and `core.eol=lf` are set via .gitattributes, they seem to have no affect when the git global config is not `core.autocrlf=false`.

So, I think I found a compromise to override a user's global config for this repo:
1. `* text autocrlf=false` prevents line ending normalization
2. `* text eol=lf` ensures workspace uses LF on checkout

This way if a CRLF is introduced, then `npm run check` will likely fail. At that point the user should be aware of the line endings' preference for this repo and convert them manually as needed.

### git docs links
This problem seems to have been addressed so many times (in various ways) that the docs are very nuanced and damn confusing.
- [core.eol](https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreeol)
- [core.autocrlf](https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreautocrlf)
- [gitattributes specs](https://git-scm.com/docs/gitattributes)